include/deal.II/matrix_free/fe_point_evaluation.h

```diff
@@ -682,40 +682,16 @@ namespace internal
 
 
 /**
- * This class provides an interface to the evaluation of interpolated solution
- * values and gradients on cells on arbitrary reference point positions. These
- * points can change from cell to cell, both with respect to their quantity as
- * well to the location. The two typical use cases are evaluations on
- * non-matching grids and particle simulations.
- *
- * The use of this class is similar to FEValues or FEEvaluation: The class is
- * first initialized to a cell by calling `FEPointEvaluation::reinit(cell,
- * unit_points)`, with the main difference to the other concepts that the
- * underlying points in reference coordinates need to be passed along. Then,
- * upon call to evaluate() or integrate(), the user can compute information at
- * the give points. Eventually, the access functions get_value() or
- * get_gradient() allow to query this information at a specific point index.
- *
- * The functionality is similar to creating an FEValues object with a
- * Quadrature object on the `unit_points` on every cell separately and then
- * calling FEValues::get_function_values or FEValues::get_function_gradients,
- * and for some elements and mappings this is what actually happens
- * internally. For specific combinations of Mapping and FiniteElement
- * realizations, however, there is a much more efficient implementation that
- * avoids the memory allocation and other expensive start-up cost of
- * FEValues. Currently, the functionality is specialized for mappings derived
- * from MappingQ and MappingCartesian and for finite elements with tensor
- * product structure that work with the
- * @ref matrixfree
- * module. In those cases, the cost implied
- * by this class is similar (or sometimes even somewhat lower) than using
- * `FEValues::reinit(cell)` followed by `FEValues::get_function_gradients`.
+ * Base class of FEPointEvaluation and FEFacePointEvaluation. This class needs
+ * usually not be called in user code and does not have any public
+ * constructor. The usage is through the class
+ * FEPointEvaluation/FEFacePointEvaluation instead.
  */
 template <int n_components_,
           int dim,
           int spacedim    = dim,
           typename Number = double>
-class FEPointEvaluation
+class FEPointEvaluationBase
```

Comment:
I think you could introduce the base class in a separate PR.

Reply:
Should be possible, I can do that if you want me to. It only makes sense after we have `FEFacePointEvaluation`. Do you mean without new FCL stuff but bringing the ECL path to `FEFacePointEvaluation` only?